astropy/coordinates/representation.py

```diff
@@ -444,54 +625,42 @@ def __init__(self, x, y=None, z=None, unit=None, xyz_axis=None, copy=True):
             raise ValueError("x, y, and z are required to instantiate {0}"
                              .format(self.__class__.__name__))
 
-        try:
-            x = self.attr_classes['x'](x, unit=unit, copy=copy)
-            y = self.attr_classes['y'](y, unit=unit, copy=copy)
-            z = self.attr_classes['z'](z, unit=unit, copy=copy)
-        except u.UnitsError:
-            raise u.UnitsError('x, y, and z should have a unit consistent with '
-                               '{0}'.format(unit))
-        except Exception:
-            raise TypeError('x, y, and z should be able to initialize ' +
-                            ('a {0}'.format(self.attr_classes['x'].__name__))
-                             if len(set(self.attr_classes.values)) == 1 else
-                            ('{0}, {1}, and {2}, resp.'.format(
-                                cls.__name__ for cls in
-                                self.attr_classes.values())))
-
-        if not (x.unit.physical_type ==
-                y.unit.physical_type == z.unit.physical_type):
-            raise u.UnitsError("x, y, and z should have matching physical types")
+        if unit is not None:
+            x = u.Quantity(x, unit, copy=copy, subok=True)
+            y = u.Quantity(y, unit, copy=copy, subok=True)
+            z = u.Quantity(z, unit, copy=copy, subok=True)
+            copy = False
 
-        try:
-            x, y, z = broadcast_arrays(x, y, z, subok=True)
-        except ValueError:
-            raise ValueError("Input parameters x, y, and z cannot be broadcast")
+        super(CartesianRepresentation, self).__init__(x, y, z, copy=copy)
+        if not (self._x.unit.physical_type ==
+                self._y.unit.physical_type == self._z.unit.physical_type):
+            raise u.UnitsError("x, y, and z should have matching physical types")
 
-        self._x = x
-        self._y = y
-        self._z = z
+    def unit_vectors(self):
```

Comment:
Sure, that's fine.  I want to argue and say we *should* do it, but I realize this is really an edge case and probably isn't worth the effort ;)